test/acceptance/PushPullNotificationsSpec.scala

```diff
@@ -16,8 +16,6 @@
 
```

Comment:
As a general comment the whole spec is becoming rather complicated WRT data structures and processing. Will you have some time next week to pair up and see how we can improve it?

Reply:
yes, sure, I tried to make it simpler by extracting bits into different traits.